### DISCUSSION TITLE
Access raw audio data in cpp from browser microphone

### DISCUSSION THREAD

**Discussion by GrigorySamokhin** [paullouisageneau/libdatachannel#750] [Q&A] [answered]
Hello! I would like to find out if it is possible to access raw audio data on the cpp side? I have a stt model that takes decoded sound in chunks and process it.
So I was wondering if I could use this library in this case.

**Comment by paullouisageneau** [comment, marked as answer]
Yes you can do so with libdatachannel. You could do something similar to the [media-receiver](https://github.com/paullouisageneau/libdatachannel/tree/master/examples/media-receiver) example, but using an audio track with OPUS codec. 

The `onMessage` callback will receive RTP packets that you could either decode with ffmpeg or gstreamer, or [depacketize](https://www.rfc-editor.org/rfc/rfc7587.html#section-4) yourself and decode with libopus.

**Reply by GrigorySamokhin** [reply]
Thanks!
